searching/exponential_search.c

```diff
@@ -0,0 +1,104 @@
+/**
+ *  \file
+ *  \brief [Exponential Search](https://github.com/TheAlgorithms/Algorithms-Explanation/blob/master/en/Search%20Algorithms/Exponential%20Search.md)
+ *  \author [Alessio Farinelli](https://github.com/faridevnz)
+ */
+#include <assert.h>  // for assert
+#include <inttypes.h>  // for int64_t, uint16_t
+
+#define ELEMENT -10
+
+int64_t binary_search(const int64_t* arr, const uint16_t l_index, const uint16_t r_index, const int64_t n);
+int64_t exponential_search(const int64_t* arr, const uint16_t length, const int64_t n);
+static void test();
+
+/**
+ *  Function: exponential_search
```

Comment:
Add a brief description of what the function does (same in other parts of the code).